src/main/java/com/facebook/openwifirrm/modules/Modeler.java

```diff
@@ -267,7 +270,10 @@ private void fetchInitialData() {
 			if (state != null) {
 				try {
 					State stateModel = gson.fromJson(state, State.class);
-					dataModel.latestState.put(device.serialNumber, stateModel);
+					dataModel.latestStates.put(
+						device.serialNumber,
+						new ArrayList<>(Arrays.asList(stateModel))
```

Comment:
1. I don't think `new ArrayList<>` is needed here
2. Don't you want to append the `latestStates` list if `serialNumber` already exists? like line 308-313.